File: frontend/src/types/media.ts
// 媒体类型（与后端枚举保持一致）
export type MediaType = 'photo' | 'video'

// 媒体项接口
export interface MediaItem {
  id: string
  name: string
  type: MediaType
  url: string
  thumbnail_url?: string // 可能不存在缩略图
  size: number
  upload_date: string // 与后端字段名一致
  width?: number
  height?: number
  duration?: number
  path?: string // 文件路径
  description?: string // 媒体内容说明
  score?: number // 搜索相似度分数
  global_media_id?: string // 32位全局唯一ID
}

// 媒体列表响应接口
export interface MediaListResponse {
  items: MediaItem[]
  total: number
  page: number
  page_size: number // 与后端字段一致
  has_more: boolean
}

// 媒体上传状态
export interface UploadStatus {
  id: string
  file: File
  progress: number
  status: 'pending' | 'uploading' | 'success' | 'error'
  error?: string
}

// 上传结果响应
export interface UploadResult {
  success: boolean
  file_name: string
  file_type: MediaType
  file_size: number
  file_path: string
  message: string
} 

File: backend/app/schemas/media.py
from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel


class MediaType(str, Enum):
    """
    媒体类型枚举
    """
    PHOTO = "photo"
    VIDEO = "video"


class MediaItem(BaseModel):
    """
    媒体项目模型
    """
    id: str  # 文件名作为ID
    name: str
    type: MediaType
    path: str
    size: int  # 文件大小（字节）
    url: str  # 访问URL
    thumbnail_url: Optional[str] = None  # 缩略图URL（仅对照片和视频有效）
    upload_date: datetime
    width: Optional[int] = None  # 宽度（像素）
    height: Optional[int] = None  # 高度（像素）
    duration: Optional[float] = None  # 视频时长（秒）
    description: Optional[str] = None  # 媒体内容说明


class MediaList(BaseModel):
    """
    媒体列表响应模型
    """
    items: List[MediaItem]
    total: int
    page: int
    page_size: int


class UploadResult(BaseModel):
    """
    上传结果响应模型
    """
    success: bool
    file_name: str
    file_type: MediaType
    file_size: int
    file_path: str
    message: Optional[str] = None
